pkg/authz/check/service.go

```diff
@@ -56,7 +59,8 @@ func (svc CheckService) getMatchingSubjects(ctx context.Context, objectType stri
 	log.Ctx(ctx).Debug().Msgf("Getting matching subjects for %s:%s#%s@___%s", objectType, objectId, relation, checkCtx)
 
 	warrantSpecs := make([]warrant.WarrantSpec, 0)
-	objectTypeSpec, err := svc.ObjectTypeSvc.GetByTypeId(ctx, objectType)
+	// TODO: caller should pass this objectTypeSpec
```

Comment:
Technically the object type we're looking up could change during the course of a check (e.g. when an object type references another, so we hop to a different type), so I don't think the caller can pass in the target object type. A better approach might be to add a cache to the object type service.